packages/babylon/src/plugins/estree.js

```diff
@@ -239,6 +238,9 @@ export default (superClass: Class<Parser>): Class<Parser> =>
       node.raw = node.extra.raw;
       delete node.extra;
 
+      const isExpression = !this.match(tt.braceL);
+      node.expression = isExpression;
```

Comment:
Could this be added in an override for `parseFunctionBody`? That's the function where you initially removed `this.expression = false`, so I could see us doing

```
parseFunctionBody(node: N.Function, allowExpression: ?boolean): void {

Reply:
Updated. Thanks

Reply:
I think `parseFunctionBody` is to generic as it would add the field also to object methods for example. The field should only show up on arrow functions according to the estree spec, so maybe we should overwrite `parseArrowExpression` instead.

```js
parseArrowExpression(
  node: N.ArrowFunctionExpression,
  params?: ?(N.Expression[]),
  isAsync?: boolean,
): N.ArrowFunctionExpression {
  node = super.parseFunctionBody(node, allowExpression);
  node.expression = node.body.type !== "BlockStatement";

  return node;
}
```

Reply:
Okay seems that is there for backwards compatibility with this https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Expression_Closures

So ignore my previous comment  :)

Can you also add the field back in `initFunction` by overwriting the function in estree.js, then the estree plugin should do the same the normal parser did before.

```js
initFunction(node: N.BodilessFunctionOrMethodBase, isAsync: ?boolean): void {
  super.initFunction(node, isAsync);
  node.expression = false;
}
```

Reply:
@danez good catch. PR updated.